Exercise 2

az-aesteban opened the pull request:
- View Lifecycle
- Navigation
- Prompts
- Popovers
- Action Sheets

az-aesteban committed:
Exercise 2

- View Lifecycle
- Navigation
- Prompts
- Popovers
- Action Sheets

az-aesteban committed:
Exercise 1 Orientation on load hack

- viewWillTransitionToSize is only called when device orientation changes
- So quick hack is to check view's width and height to check orientation

az-char commented on IOSExercises/Classes/LocaleManager.m:
```diff
@@ -12,12 +12,14 @@
 - (NSString *)languageFromRegion {
+    NSLog(@"%@", [[NSLocale currentLocale] countryCode]);
     NSString *countryCode = [[NSLocale currentLocale] countryCode];
     NSString *language;
```
Always initialize on declare. In this case, to `nil`

az-char commented on IOSExercises/Views/DetailsViewController.m:
```diff
@@ -0,0 +1,341 @@
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    [self initDefaultValue];
```
Avoid reserved keywords like `init` (since `init` signifies that an instance is being created). In this case, `setup` seems to be the better prefix

az-char commented on IOSExercises/Views/DetailsViewController.m:
```diff
@@ -0,0 +1,341 @@
+- (void)showColorPicker:(UITapGestureRecognizer *)recognizer {
+    if (UIUserInterfaceIdiomPhone == [[UIDevice currentDevice] userInterfaceIdiom]) {
+        /*
+         * Display as action sheet
```
For multiline comments inside a method, please use `// ` instead (not industry-standard, but as a project standard)

az-char commented on IOSExercises/Views/DetailsViewController.m:
```diff
@@ -0,0 +1,341 @@
+         */
+        [self displayColorPickerActionSheet];
+    } else {
+        [self popoverColorPicker];
```
In Obj-C, "popover" is usually used as a noun, and for methods that "do", they should start with a verb.

In this case, `displayColorPickerPopover`

az-char commented on IOSExercises/Views/DetailsViewController.m:
```diff
@@ -0,0 +1,341 @@
+}
+
+- (void)showDatePicker:(UITapGestureRecognizer *)recognizer {
+    [self popoverDateTimePicker];
```
Log user actions

az-char commented on IOSExercises/Views/DetailsViewController.m:
```diff
@@ -0,0 +1,341 @@
+    [self popoverDateTimePicker];
+}
+
+- (IBAction)greetIfBirthday:(id)sender {
```
IBActions should be named after the action that was done

Then, the handling should be named after what it attempts to do (which is what was done here)

az-char commented on IOSExercises/Views/DetailsViewController.m:
```diff
@@ -0,0 +1,341 @@
+
+    NSDate *birthdate = [dateFormatter dateFromString:self.birthdayLabel.text];
+
+    NSDateComponents *bdayComponent = [[NSCalendar currentCalendar] components:NSCalendarUnitDay | NSCalendarUnitMonth
```
Avoid using shorthand forms of words in variable names as a project-standard (we've had issues before where shortened forms of variables made them unreadable and confusing, especially when domain owner left the project already)

az-char commented on IOSExercises/Views/DetailsViewController.m:
```diff
@@ -0,0 +1,341 @@
+    [self popoverWithGreeting:greeting];
+}
+
+- (IBAction)chooseColorRed:(id)sender {
```
`didChoose...`

az-char commented on IOSExercises/Views/DetailsViewController.m:
```diff
@@ -0,0 +1,341 @@
+    [self.colorLabel setText:@"Black"];
+}
+
+- (IBAction)saveBirthday:(id)sender {
```
`didTap...`

az-char commented on IOSExercises/Views/DetailsViewController.m:
```diff
@@ -0,0 +1,341 @@
+
+#pragma mark - Popover View Controller Methods
+
+- (void)popoverDateTimePicker{
```
Spacing

az-char commented on IOSExercises/Views/DetailsViewController.m:
```diff
@@ -0,0 +1,341 @@
+    self.popoverViewController.view = [[UIView alloc] init];
+    [self.popoverViewController setPreferredContentSize:CGSizeMake(250.f, 270.f)];
+
+    //[self setBirthday];
```
Remove unnecessary code

az-char commented on IOSExercises/Views/DetailsViewController.m:
```diff
@@ -0,0 +1,341 @@
+    [self.datePicker setDatePickerMode:UIDatePickerModeDate];
+    [self.popoverViewController.view addSubview:self.datePicker];
+
+    UIButton *saveButton = [UIButton buttonWithType:UIButtonTypeSystem];
```
Refactor method when possible (and just group the helper methods under a #pragma mark, for example)

az-char commented on IOSExercises/Views/DetailsViewController.m:
```diff
@@ -0,0 +1,341 @@
+}
+
+- (void)popoverWithGreeting:(NSString *)aGreeting {
+
```
Have care in adding extra (empty) lines in code. It isn't a concern now, but once the class starts to bloat (due to increasing features), the extra lines mean less lines in the IDE's view (so make sure you really want them, eg for line grouping in terms of logic, param readability, and so on)

az-char commented on IOSExercises/Views/DetailsViewController.m:
```diff
@@ -0,0 +1,341 @@
+    [self.birthdayLabel setText:[dateFormatter stringFromDate:aDate]];
+}
+
+- (UIAlertAction *)actionChooseColor:(NSString *)aTitle {
```
`alertActionForChoosingColor:`

az-char commented on IOSExercises/Views/ProfileViewController.m:
```diff
@@ -31,9 +34,27 @@ - (void)viewDidLoad {
         [self activateIPhoneConstraints];
     }
+
+    UIDeviceOrientation orientation = self.view.bounds.size.width > self.view.bounds.size.height ? UIDeviceOrientationLandscapeRight: UIDeviceOrientationPortrait;
```
Spacing

`condition ? if yes : if no`

az-char commented on IOSExercises/Views/ProfileViewController.m:
```diff
@@ -31,9 +34,27 @@ - (void)viewDidLoad {
+                                                                        bundle:[NSBundle mainBundle]];
+    }
+
+    self.detailsViewController.title = @"Hello, hello!"; // later dapat value ni sa profile view label
```
Comments in English (project-standard, for our counterparts from other countries)

az-char commented:
Done reviewing, good job so far! Minor comments, mostly refactoring-related due to constraints/view in code

az-char commented:
Discussion notes:
1- Please add a border to the image and caption label
2- Please use NSLayoutAnchor for iPhone constraints
3- Present "It's your birthday" screen as a modal
4- Implement Save button

az-char commented on IOSExercises/Views/DetailsViewController.m:
```diff
@@ -0,0 +1,340 @@
+
+-(NSDateFormatter *)birthdayFormatter {
+    NSDateFormatter *dateFormatter = [[NSDateFormatter alloc] init];
+    [dateFormatter setDateFormat:@"MMMM d"];
```
Make the date format a constant: `static NSString * const kBirthdayDateFormat = @"MMMM d";`

This line comes after the import statements and before the `@interface` declaration for the private category

az-aesteban committed:
Apply code comments

- Naming conventions, formatting, refactor
- Add a border to the image and caption label
- Use NSLayoutAnchor for iPhone constraints
- Present "It's your birthday" screen as a modal
- Implement Save button
